iothub/service/src/Query/QueryClient.cs

```diff
@@ -170,41 +154,94 @@ await _internalRetryHandler
+                }
 
-                if (!string.IsNullOrWhiteSpace(options?.ContinuationToken))
+                async Task<Page<ScheduledJob>> firstPageFunc(int? pageSizeHint)
                 {
-                    request.Headers.Add(ContinuationTokenHeader, options?.ContinuationToken);
+                    cancellationToken.ThrowIfCancellationRequested();
+                    using HttpRequestMessage request = _httpRequestMessageFactory.CreateRequest(
+                                        HttpMethod.Get,
+                                        s_jobsQueryFormat,
+                                        _credentialProvider,
+                                        null,
+                                        BuildQueryJobQueryString(options));
+
+                    return await BuildAndSendRequest<ScheduledJob>(request, null, pageSizeHint, cancellationToken);
                 }
 
-                if (options?.PageSize != null)
+                return PageableHelpers.CreateAsyncEnumerable(firstPageFunc, nextPageFunc);
+            }
+            catch (HttpRequestException ex)
+            {
+                if (Fx.ContainsAuthenticationException(ex))
                 {
-                    request.Headers.Add(PageSizeHeader, options.PageSize.ToString());
+                    throw new IotHubServiceException(ex.Message, HttpStatusCode.Unauthorized, IotHubServiceErrorCode.IotHubUnauthorizedAccess, null, ex);
                 }
+                throw new IotHubServiceException(ex.Message, HttpStatusCode.RequestTimeout, IotHubServiceErrorCode.Unknown, null, ex);
+            }
+            catch (Exception ex) when (Logging.IsEnabled)
+            {
+                Logging.Error(this, $"Creating query with jobType: {options?.JobType}, jobStatus: {options?.JobStatus} threw an exception: {ex}", nameof(CreateAsync));
+                throw;
+            }
+            finally
+            {
+                if (Logging.IsEnabled)
+                    Logging.Exit(this, $"Creating query with jobType: {options?.JobType}, jobStatus: {options?.JobStatus}", nameof(CreateAsync));
+            }
+        }
+
+        private async Task<Page<T>> BuildAndSendRequest<T>(HttpRequestMessage request, string continuationToken, int? pageSizeHint, CancellationToken cancellationToken)
+        {
+            if (!string.IsNullOrWhiteSpace(continuationToken))
+            { 
+                request.Headers.Add(ContinuationTokenHeader, continuationToken);
+            }
+
+            if (pageSizeHint != null)
+            {
+                request.Headers.Add(PageSizeHeader, pageSizeHint.ToString());
+            }
+
+            if (request.Content != null)
+            { 
+                request.Content.Headers.ContentType = new MediaTypeHeaderValue("application/json") 
+                { 
+                    CharSet = "utf-8" 
+                };
+            }
```

Comment:
```suggestion
            (request.Content?.Headers.?ContentType ??= new MediaTypeHeaderValue("application/json") 
            { 
                CharSet = "utf-8" 
            };
            }
```

Reply:
This actually didn't work for me. Even after replacing ```Headers.?``` with ```Headers?.```

Reply:
Oh well.